File: App008_CardView/app/src/main/java/com/example/app08_cardview/GameAdapter.java
package com.example.app08_cardview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class GameAdapter extends RecyclerView.Adapter<GameAdapter.ViewHolder> {

    // 1 -- Data:
    private Context context;
    private ArrayList<GameModel> dataList;

    // 2 -- Constructor
    public GameAdapter(Context context, ArrayList<GameModel> dataList) {
        this.context = context;
        this.dataList = dataList;
    }

    // 3 -- View Holder
    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView gameImage;
        private TextView gameTitle;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            gameImage = itemView.findViewById(R.id.card_image);
            gameTitle = itemView.findViewById(R.id.card_title);
        }
    }

    // 4 -- Implement Methods
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.card_item, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        GameModel model = dataList.get(position);
        holder.gameTitle.setText(model.getName());
        holder.gameImage.setImageResource(model.getImage());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(
                        context,
                        "Title: " + model.getName(),
                        Toast.LENGTH_SHORT
                ).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }
}

File: App030_Kt_CardView/app/src/main/java/com/kes/app030_kt_cardview/GameAdapter.kt
package com.kes.app030_kt_cardview

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.kes.app030_kt_cardview.models.Game

class GameAdapter(private val games: ArrayList<Game>): RecyclerView.Adapter<GameAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.card_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val game = games[position]
        holder.imageView.setImageResource(game.imageSrc)
        holder.titleView.text = game.name
    }

    override fun getItemCount(): Int = games.size

    inner class ViewHolder(view: View): RecyclerView.ViewHolder(view) {
        val imageView: ImageView = view.findViewById(R.id.card_image)
        val titleView: TextView = view.findViewById(R.id.card_text)
        
        init {
            view.setOnClickListener() {
                Toast.makeText(
                    view.context,
                    games[adapterPosition].name,
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }
}